### PR TITLE
[FIX] mail: fix partner suggestion order

### PR DESCRIPTION
**Pull request by phenix-factory**
Before this PR, suggestions when mentioning partners would not put internal user that are also followers at the top of the list. This PR fix the issue by back-porting record equality code from master.

Task-3472727

### CODE REVIEW

File: addons/mail/static/tests/suggestion/suggestion_tests.js [hunk 1]
```diff
@@ -215,3 +215,72 @@ QUnit.test("Suggestions are shown after delimiter was used in text (#)", async (
...
+        },
+        {
+            email: "partner@example.com",
+            name: "Trigger X simple partner",
+        },
+        {
+            email: "apartner@example.com",
+            name: "Trigger X simple partner",
+        },
+    ]);
```
Comment (alexkuhn) [inline addons/mail/static/tests/suggestion/suggestion_tests.js:253]: Keep name short, it's hard to read test when the name are sentences.
`A B C D` or `John James Jimmy Demo`, as long as this is short and easy to read

File: addons/mail/static/tests/suggestion/suggestion_tests.js [hunk 2]
```diff
@@ -215,3 +215,72 @@ QUnit.test("Suggestions are shown after delimiter was used in text (#)", async (
...
+    );
+    assert.strictEqual(suggestions[1].textContent, "Trigger C internal user(Internal@example.com)");
+    assert.strictEqual(suggestions[2].textContent, "Trigger B follower(follower@example.com)");
+    assert.strictEqual(suggestions[3].textContent, "Trigger A simple partner(partner@example.com)");
+    assert.strictEqual(
+        suggestions[4].textContent,
+        "Trigger X simple partner(apartner@example.com)"
+    );
+    assert.strictEqual(suggestions[5].textContent, "Trigger X simple partner(partner@example.com)");
+    assert.strictEqual(suggestions[6].textContent, "Trigger Y simple partner(partner@example.com)");
```
Comment (alexkuhn) [inline addons/mail/static/tests/suggestion/suggestion_tests.js:285]: If you have shorter name and email, each of them is a single line

**Review by alexkuhn — APPROVED** [review]
@robodoo r+
@fw-bot up-to saas-16.4

**Comment by alexkuhn** [comment]
@robodoo retry

**Comment by phenix-factory** [comment]
@robodoo retry

**Pull request closed without merging**